tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -54,7 +55,11 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
 			// Delete the pods created. This should idealy be done by `odo
 			// service delete` but that's not implemented for operator backed
 			// services yet.
-			helper.CmdShouldPass("oc", "delete", "EtcdCluster", "example")
+			// Wait for the pods of EtcdCluster service to get cleared off
+			// cluster. This is to avoid CI flakes because in next test, we're
+			// again doing similar regex check which picks up terminating pod
+			// from this run instead of initiating pod from that run
+			helper.CmdShouldRunWithTimeout(time.Duration(15)*time.Second, "oc", "delete", "--wait", "EtcdCluster", "example")
```

Comment:
I've added this sleep because I found a flake in local environment. It happens when we move to next test spec. Doing `oc delete EtcdCluster/example` doesn't wait for the underlying pods to be deleted. So in the next spec, when we are doing some match with regex, it might pick-up a pod which is under termination from this run instead of picking up a pod which is initiation from that run.

Another thing we could possibly do is to modify the regex we're matching for. But `regexp.MustCompile(`example-.[a-z0-9]*Init*`).FindString(pods)` doesn't yield what I'm looking for which is the pod name.